### PR TITLE
CORE-5114 serialization tweaks

### PR DESCRIPTION
**Pull request by vlajos**
* Simplify SandboxFactory and SandboxManagementService related integration tests a bit.
* ConsensualSignedTransactionSerializationFlow and its smoke test.
* Add ledger serialization modules to flow-processor.
* Add corda-ledger-consensual to db-processor.
* Few small tweaks.

### CODE REVIEW

File: testing/sandboxes/src/main/kotlin/net/corda/testing/sandboxes/EmptySandboxService.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    @Reference
+    private val sandboxCreationService: SandboxCreationService
+) {
+    val emptySandboxGroup: SandboxGroup = sandboxCreationService.createSandboxGroup(emptyList())
+
+    @Suppress("unused")
+    @Deactivate
+    fun cleanup() {
+        sandboxCreationService.unloadSandboxGroup(emptySandboxGroup)
+    }
```
Comment (chrisr3) [inline testing/sandboxes/src/main/kotlin/net/corda/testing/sandboxes/EmptySandboxService.kt]: Why do we want a general component that can create an empty sandbox? I think this should stay with the project that has a use-case for it.

Comment (vlajos) [reply testing/sandboxes/src/main/kotlin/net/corda/testing/sandboxes/EmptySandboxService.kt]: @chrisr3 
I needed this in 4 different modules so far.
Or are there maybe a nicer solution to have an empty/basic sandbox just to bootstrap serialization?

Comment (chrisr3) [reply testing/sandboxes/src/main/kotlin/net/corda/testing/sandboxes/EmptySandboxService.kt]: If your use-case is "bootstrapping serialization" - which isn't at all obvious - then I think we _need_ a nicer solution.

How does creating an empty sandbox group "bootstrap serialization" anyway? That alone strikes me as odd :thinking:. Also, _why_ does serialization even need bootstrapping in the first place? Does Corda do anything like this?

Comment (vlajos) [reply testing/sandboxes/src/main/kotlin/net/corda/testing/sandboxes/EmptySandboxService.kt]: @chrisr3 
Maybe "bootstrapping" is not the best word.
I am referring to these usages:
https://github.com/corda/corda-runtime-os/blob/vlajos/CORE-5114/serialization-tweaks/libs/ledger/ledger-common-impl/src/integrationTest/kotlin/net/corda/ledger/common/impl/transaction/serializer/test/WireTransactionKryoSerializationTest.kt#L72
https://github.com/corda/corda-runtime-os/blob/vlajos/CORE-5114/serialization-tweaks/libs/ledger/ledger-common-serialization/src/integrationTest/kotlin/net/corda/ledger/common/transaction/serialization/test/WireTransactionAMQPSerializationTest.kt#L96

Comment (chrisr3) [reply testing/sandboxes/src/main/kotlin/net/corda/testing/sandboxes/EmptySandboxService.kt]: The problem is that we expect sandboxes to be created from CPIs, so having a component _specifically_ to create a sandbox that has no bundles at all is an anomaly.

I think PR #2230 will help to implement a _slightly_ more general case. The idea is:
```kotlin
private lateinit var emptySandboxGroup: SandboxGroup
```
and then:
```kotlin
lifecycle.accept(sandboxSetup) { setup ->
    val sandboxCreationService = setup.fetchService<SandboxCreationService>(timeout = 1500)
    emptySandboxGroup = sandboxCreationService.createSandboxGroup(emptyList())
    setup.withCleanup {
        sandboxCreationService.unloadSandboxGroup(emptySandboxGroup)
    }
    ...
}
```
It doesn't bother me if this code is repeated a few times because I'm expecting us to need tests containing _actual_ CPIs at some point :crystal_ball:.

Comment (vlajos) [reply testing/sandboxes/src/main/kotlin/net/corda/testing/sandboxes/EmptySandboxService.kt]: Thanks.
The original examples what I found and used as a starting point:
https://github.com/corda/corda-runtime-os/tree/vlajos/CORE-5114/serialization-tweaks/libs/serialization/serialization-amqp/src/integrationTest/kotlin/net/corda/serialization/amqp/test
https://github.com/corda/corda-runtime-os/tree/vlajos/CORE-5114/serialization-tweaks/libs/serialization/serialization-kryo/src/integrationTest/kotlin/net/corda/kryoserialization/test
worked with CPI based sandboxes.
But then I realized that for testing what I wanted to test, I do not need the CPIs at all, then stripped back the tests and I ended up having these empty Sandboxes.

If it makes our life easier, I can just put back the unused CPIs. But that does not sound very clean either.

Re https://github.com/corda/corda-runtime-os/pull/2230 : How can I use that? I am a bit lost...

Comment (chrisr3) [reply testing/sandboxes/src/main/kotlin/net/corda/testing/sandboxes/EmptySandboxService.kt]: Heh, someone needs to approve that PR first... :grinning:.

Comment (vlajos) [reply testing/sandboxes/src/main/kotlin/net/corda/testing/sandboxes/EmptySandboxService.kt]: I could just cherry-pick it. ;-)

Comment (chrisr3) [reply testing/sandboxes/src/main/kotlin/net/corda/testing/sandboxes/EmptySandboxService.kt]: > If it makes our life easier, I can just put back the unused CPIs. But that does not sound very clean either.

I actually thought you'd need CPIs so that you'd have sandbox classes to serialize in the first place... :thinking:

Comment (chrisr3) [reply testing/sandboxes/src/main/kotlin/net/corda/testing/sandboxes/EmptySandboxService.kt]: #2230 has now been merged, so you can rebase.

Comment (vlajos) [reply testing/sandboxes/src/main/kotlin/net/corda/testing/sandboxes/EmptySandboxService.kt]: Thanks. I have applied that pattern.

**Review by mnesbit — APPROVED** [review]
I don't want to block anybody, so this is fine. However, I think we need to arrange our platform modules carefully to keep DB worker dependencies small and specific versus the ones pulled into flow engine for system flows

**Review by blsemo — APPROVED** [review]
Looks good to me

**Review by ronanbrowne — APPROVED** [review]
LGTM (Build)
